### PR TITLE
feat(destination): onboard lemnisk integration

### PR DESCRIPTION
**Pull request by anantjain45823**
## PR Description

Onboarding new Integration Lemnisk Marketing automation continued from #806 
## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by anantjain45823** [comment]
SonarCloud is failing upon the script that Lemnisk has written. Shall we do something about this?

**Comment by yashasvibajpai** [comment]
> SonarCloud is failing upon the script that Lemnisk has written. Shall we do something about this?

Check with sdk team can we selectively skip the check here?

File: src/integrations/Lemnisk/constants.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const NAME = 'LEMNISK';
+const CNameMapping = {
+  [NAME]: NAME,
+  'LEMNISK_MARKETING_AUTOMATION': NAME,
+  'Lemnisk Marketing Automation': NAME,
+  'LemniskMarketingAutomation': NAME,
+  'lemniskmarketingautomation': NAME,
+  'lemniskMarketingAutomation': NAME,
+  'lemnisk': NAME,
```
Comment (MoumitaM) [inline src/integrations/Lemnisk/constants.js:9]: Add `Lemnisk`

File: src/utils/client_server_name.js [hunk 2]
```diff
@@ -70,6 +70,7 @@ const clientToServerNames = {
   MICROSOFT_CLARITY: 'Microsoft Clarity',
   SENDINBLUE: 'Sendinblue',
   OLARK: 'Olark',
+  Lemnisk: "Lemnisk",
```
Comment (MoumitaM) [inline src/utils/client_server_name.js:73]: This is the mapping from the config name to the display name.
As you have declared `LEMNISK` as the config name in the constants.js file then the mapping should be 
`LEMNISK: "Lemnisk",`

File: src/utils/config_to_integration_names.js [hunk 3]
```diff
@@ -72,6 +72,7 @@ const configToIntNames = {
   MICROSOFT_CLARITY: 'MicrosoftClarity',
   SENDINBLUE: 'Sendinblue',
   OLARK: 'Olark',
+  Lemnisk: "Lemnisk",
```
Comment (MoumitaM) [inline src/utils/config_to_integration_names.js:75]:
```suggestion
  LEMNISK: "Lemnisk",
```

File: __tests__/integrations/Lemnisk/lemnisk.test.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
+import LemniskMarketingAutomation from "../../../src/integrations/Lemnisk/browser";
+
+beforeAll(() => { });
```
Comment (MoumitaM) [inline __tests__/integrations/Lemnisk/lemnisk.test.js:3]: Remove unnecessary line no 3

**Review by MoumitaM — CHANGES_REQUESTED** [review]

File: __tests__/integrations/Lemnisk/lemnisk.test.js (hunk 4 above)
Comment (anantjain45823) [reply __tests__/integrations/Lemnisk/lemnisk.test.js]: Done

File: src/utils/config_to_integration_names.js (hunk 3 above)
Comment (anantjain45823) [reply src/utils/config_to_integration_names.js]: Done

File: src/utils/client_server_name.js (hunk 2 above)
Comment (anantjain45823) [reply src/utils/client_server_name.js]: Done

File: src/integrations/Lemnisk/constants.js (hunk 1 above)
Comment (anantjain45823) [reply src/integrations/Lemnisk/constants.js:9]: Done

**Review by MoumitaM — APPROVED** [review]

File: src/integrations/Lemnisk/browser.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        logger.debug('===In Lemnisk Marketing Automation identify===');
+        const userId = rudderElement.message.userId || rudderElement.message.anonymousId;
+        if (!userId) {
+            logger.debug('[LemniskMarketingAutomation] identify:: user id is required');
+            return;
+        }
+        // disabling eslint as message will be there iinn any case
+        // eslint-disable-next-line no-unsafe-optional-chaining 
+        const { traits } = rudderElement.message?.context;
+        window.lmSMTObj.identify(rudderElement.message.userId, traits);
```
Comment (ItsSudip) [inline src/integrations/Lemnisk/browser.js:77]:
```suggestion
        const { traits } = rudderElement.message?.context;
        if(traits){
        window.lmSMTObj.identify(rudderElement.message.userId, traits);
        }
        else
        {
        window.lmSMTObj.identify(rudderElement.message.userId);
        }
```

**Review by ItsSudip — DISMISSED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**